crates/rpc/eth/block.rs

```diff
@@ -193,6 +198,43 @@ impl RpcHandler for GetBlockReceiptsRequest {
     }
 }
 
+impl RpcHandler for GetRawHeaderRequest {
+    fn parse(params: &Option<Vec<Value>>) -> Result<GetRawHeaderRequest, RpcErr> {
+        let params = params.as_ref().ok_or(RpcErr::BadParams)?;
+        if params.len() != 1 {
+            return Err(RpcErr::BadParams);
+        };
+
+        let block: BlockIdentifier = serde_json::from_value(params[0].clone())?;
+        let block_str: String = serde_json::from_value(params[0].clone())?;
+
+        if let BlockIdentifier::Number(_) = block {
+            if !block_str.starts_with("0x") {
+                return Err(RpcErr::BadHexFormat(0));
+            }
+        }
```

Comment:
This looks like a check that should be done for all cases where `BlockIdentifier` is used. Maybe we could add a method for `BlockIndentifier` that takes care of this check while parsing (such as `parse(serde_value: Value) -> Result<BlockIdentifier, RpcErr>`) that we can use for all cases without repeating code